docs/DeveloperGuide.md

```diff
@@ -2,33 +2,166 @@
+
+### Api component
+
+![Structure of the Api Component](diagrams/ApiClass.png)
+
+The `Api`,
+
+* receives a raw query from Find Command, query will contain filter conditions.
+* creates a connection to remote server containing data on resale flats.
+* formats raw query into valid query to be sent as a GET request.
+* updates all flats matching filter conditions in SearchedUnits class of Data component.
+
+**How classes within Api component interact with each other**
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where a Find command is executed.
+
+<img src="diagrams/ApiSequence.png" />
+
+### Storage component
+
+![Structure of the Storage Component](diagrams/StorageClass.png)
+
+The `Storage`,
+
+* handles read and write of units into a local text file.
+* manages data in ShortList.
+
+**How classes within Storage component interact with each other**
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where a Save command is executed.
+
+<img src="diagrams/StorageSequence.png" />
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+Low-to-middle income single users looking to buy resale flats.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+Easily find and bookmark resale flats available matching user's preference.
 
 ## User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|user|find units by location, flat type and lease remaining|search for resale flats matching my preferences|
+|v2.0|new user|see usage instructions|understand all the available commands|
+|v2.0|returning user|bookmark potential flats|refer to them in the future|
+|v2.0|user|sort flats by price, in either ascending or descending order|view flats matching my budget|
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2.  Requires internet connection.
+3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
 ## Glossary
 
-* *glossary item* - Definition
+* **Mainstream OS**: Windows, Linux, Unix, OS-X
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+Given below are instructions to test the app manually.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** These instructions only provide a starting point for testers to work on;
```

Comment:
![image](https://user-images.githubusercontent.com/77402245/113235653-194c0080-92d6-11eb-8e6b-a12244af6cdc.png)

Should it show an emoji? Perhaps you could enable the emoji at the config.yml file